dll/win32/shell32/CDefView.cpp

```diff
@@ -1511,10 +1527,10 @@ HRESULT CDefView::InvokeContextMenuCommand(CComPtr<IContextMenu>& pCM, LPCSTR lp
     cmi.hwnd = m_hWnd;
     cmi.lpVerb = lpVerb;
 
-    if (GetKeyState(VK_SHIFT) & 0x8000)
```

Comment:
What's wrong with `0x8000`?

Reply:
Comparing a value to 0 is better than testing if a bit is set.

Reply:
However, MS docs determine the returned value by "which bit is set".
[https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeystate](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeystate)

The example also uses `0x8000` to check if the key is pressed.
[https://learn.microsoft.com/en-us/windows/win32/inputdev/using-keyboard-input](https://learn.microsoft.com/en-us/windows/win32/inputdev/using-keyboard-input)

Reply:
> MS docs determine the status by checking "which bit is set"

To be pedantic; the docs describe the 4 possible return values in terms of how the bits are set, not how the caller should check them.

Surely it is no coincidence that the function returns a signed number and the state you are most interested in just happens to be the sign bit?

Reply:
https://godbolt.org/z/f5G3bW7EW

Reply:
> https://godbolt.org/z/f5G3bW7EW

You should not let the compiler see the functions you are calling there.

Anyway, GCC is smart enough to optimize checking bit 0x8000 but MSVC is not:

https://godbolt.org/z/f3v67dddT

```
call    short GetKeyState(int)                    ; GetKeyState
cwde
test    eax, eax
jge ...
```

vs 

```
call    short GetKeyState(int)                    ; GetKeyState
cwde
and     eax, 32768                                ; 00008000H
test    eax, eax
je ...
```

Reply:
> Surely it is no coincidence that the function returns a signed number and the state you are most interested in just happens to be the sign bit?

I think following the docs is better than relying on assumptions.

Reply:
> I think following the docs is better than relying on assumptions.

Which part of the GetKeyState documentation is the code not following? It checks the bit we are interested in.

[Here the OldNewThing checks with `< 0`](https://devblogs.microsoft.com/oldnewthing/20040924-00/?p=37753) for the exact CMIC flags we are talking about...

Reply:
If it's really better for MSVC and doesn't break anything, fine then.